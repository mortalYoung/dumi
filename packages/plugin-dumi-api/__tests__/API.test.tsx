import React from 'react';
import { render } from '@testing-library/react';
import API from '../src/API';

jest.mock('@@/dumi/apis', () => {
  return {
    Hello: {
      default: [
        {
          identifier: 'className',
          description: 'Extra CSS className for this component',
          'description.zh-CN': '组件额外的 CSS className',
          type: 'string',
        },
        {
          identifier: 'type',
          description: "I'm required",
          'description.zh-CN': '我是一个必选属性',
          type: 'string',
          required: true,
        },
      ],
    },
  };
});

describe('API component', () => {
  test('Match snapshot', () => {
    const { asFragment } = render(<API hideTitle={false} identifier="Hello" export="default" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
