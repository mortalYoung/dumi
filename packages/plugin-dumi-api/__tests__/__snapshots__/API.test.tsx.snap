// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API component Match snapshot 1`] = `
<DocumentFragment>
  <table
    style="margin-top: 24px;"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Type
        </th>
        <th>
          Default
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          className
        </td>
        <td>
          Extra CSS className for this component
        </td>
        <td>
          <code>
            string
          </code>
        </td>
        <td>
          <code>
            --
          </code>
        </td>
      </tr>
      <tr>
        <td>
          type
        </td>
        <td>
          I'm required
        </td>
        <td>
          <code>
            string
          </code>
        </td>
        <td>
          <code>
            (required)
          </code>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
